components/bms_boss/src/BMS.c

```diff
@@ -138,12 +143,26 @@ static void BMS100Hz_PRD(void)
                 float32_t chg_voltage = 0.0f;
                 const bool mc_valid = (CANRX_get_signal(VEH, PM100DX_tractiveSystemVoltage, &ts_voltage) == CANRX_MESSAGE_VALID);
                 const bool chg_valid = (CANRX_get_signal(VEH, BRUSA513_dcBusVoltage, &chg_voltage) == CANRX_MESSAGE_VALID);
+                
+                if (SYS.contacts == SYS_CONTACTORS_PRECHARGE && !precharge_timer_started)
+                {
+                    drv_timer_init(&precharge_timer);
+                    drv_timer_start(&precharge_timer, 0);
```

Comment:
The timer should be started above in 
```
if (SYS.contacts == SYS_CONTACTORS_OPEN)
{
    SYS_SFT_cycleContacts();
    <here>
}
```

Reply:
Also this timer will expire immediately. You are manually implementing things provided to you in the driver. This should be 
```suggestion
                    drv_timer_start(&precharge_timer, PRECHARGE_MIN_TIME_MS);
```